vwifi.c

```diff
@@ -506,6 +548,54 @@ static void inform_bss(struct vwifi_vif *vif)
     }
 }
 
+/* Helper function that prepares a structure with self-defined BSS information
+ * and "informs" the kernel about the "new" Independent BSS.
+ */
+static void ibss_inform_bss(struct vwifi_vif *vif)
+{
+    struct vwifi_vif *ibss;
+
+    list_for_each_entry (ibss, &vwifi->ibss_list, ibss_list) {
+        struct cfg80211_bss *bss = NULL;
+        struct cfg80211_inform_bss data = {
+            /* the only channel */
+            .chan = &ibss->wdev.wiphy->bands[NL80211_BAND_2GHZ]->channels[0],
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)
+            .scan_width = NL80211_BSS_CHAN_WIDTH_20,
+#endif
+            .signal = DBM_TO_MBM(rand_int_smooth(-100, -30, jiffies)),
+        };
+        int capability = WLAN_CAPABILITY_IBSS;
+
+        if (ibss->ibss_privacy)
+            capability |= WLAN_CAPABILITY_PRIVACY;
+
+        pr_info("vwifi: %s performs scan, found %s (SSID: %s, BSSID: %pM)\n",
+                vif->ndev->name, ibss->ndev->name, ibss->ssid, ibss->bssid);
+        pr_info("cap = %d, beacon_ie_len = %d\n", capability,
+                ibss->beacon_ie_len);
+
+        /* Using the CLOCK_BOOTTIME clock, which remains unaffected by changes
+         * in the system time-of-day clock and includes any time that the
+         * system is suspended.
+         * This clock is suitable for synchronizing the machines in the BSS
+         * using tsf.
+         */
+        u64 tsf = div_u64(ktime_get_boottime_ns(), 1000);
+
+        /* It is possible to use cfg80211_inform_bss() instead. */
+        bss = cfg80211_inform_bss_data(
+            vif->wdev.wiphy, &data, CFG80211_BSS_FTYPE_UNKNOWN, ibss->bssid,
```

Comment:
What if the `ibss->bssid` is generated by the system (randomized BSSID)? I'm writing because I didn't see the code assigning this member (if the "if condition" in vwifi:2079 fails).

Reply:
Indeed, generating a  BSSID should be implemented by the driver. Perhaps we need a global structure to manage the BSSID.